woocommerce/woocommerce-ios
# Add note_full_data parsing from notification payload, open proper order from notification

Closes: #13573


## Description


We noticed that iOS notification payloads were missing `note_full_data` parameter which in a combination of getting the same `note_id` for all new orders notification resulted in a bug where users would tap the new order notification and it would take them to the last order and not the one the notification was for.

Solution was to start sending the `note_full_data` parameter from backend and for client side to parse and use it.

So with this PR the app parses the `note_full_data` parameter (if there is data for it) and if it has data for the specific order we show the order.

The only thing I have left todo is marking the notification as read on watch and in notification extension. Since bunch of code/actions are not available from watch and notif extension I have some issues with that. Can we do that part in a separate PR? @Ecarrion 

## Testing information


Testing should be done in **3 parts**:

1. General notification opening for the iOS app
- Get few new orders for one of the stores you have so it triggers notifications for it. It is important to have more of them so you can open order that is not the last.
- Tap on the notification for the order that was not the last order
- Check that the app opens the proper order
- Tap on the notification for the last order
- Check that the app opens the proper order and that the notification is marked as read also on the web

https://github.com/user-attachments/assets/4e2edf8e-9b73-4244-bb23-d93361f23c6d

2. General notification opening for the watchOS app
- Do the same steps from part 1 but open notifications from your Apple Watch

https://github.com/user-attachments/assets/80865274-ad87-4cee-a45b-ea1f198e3517

3. Notification extension
- Get a new ordera for one of the stores you have so it triggers notification for it.
- Long press the notification and check the notification preview is shown correctly

![IMG_723E898202D3-1](https://github.com/user-attachments/assets/f0185550-417e-4ddb-ba86-bda2249c8d01)

## Screenshots


- Videos and screenshots are above with testing instructions

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
